### PR TITLE
[xtensa] Fixed xtensa simd correctness testing

### PR DESCRIPTION
**Pull request by Aelphy**
This is the starting point. Now `HL_TARGET=arm-64-android-tune_generic-xtensa .//test/correctness/correctness_simd_op_check_xtensa "*" gen_tests/` returns ok.

### CODE REVIEW

**Comment by steven-johnson** [comment]
Commenting out failing tests normally requires an accompanying GitHub issue to track it (mentioned in a comment), so future code readers know why. (If this is a very-temporary commenting out then maybe ok to skip.)

**Comment by Aelphy** [comment]
I will add to this PR, such that the tests turn green :)

File: test/correctness/simd_op_check_xtensa.cpp [hunk 1]
```diff
@@ -140,11 +140,11 @@ class SimdOpCheckXtensa : public SimdOpCheckTest {
         // These are not generated right now, because vectors are split now, so comment out for now.
         // Narrowing with shifting.
         // check("halide_xtensa_narrow_with_shift_i16", vector_width / 2, i16(i32_1 >> i32_2));
-        // check("halide_xtensa_narrow_with_shift_i16", vector_width / 2, i16(i32_1 / 4));
+        check("halide_xtensa_narrow_with_shift_i16", vector_width / 2, i16(i32_1 / 4));
         // check("halide_xtensa_narrow_with_shift_u16", vector_width / 2, u16(i32_1 >> i32_2));
-        // check("halide_xtensa_narrow_with_shift_u16", vector_width / 2, u16(i32_1 / 4));
+        check("halide_xtensa_narrow_with_shift_u16", vector_width / 2, u16(i32_1 / 4));
 
-        // check("IVP_AVGshouldhavefailedRNX16", vector_width / 2, i16((i32(i16_1) + i32(i16_2) + 1) / 2));
```
Comment (Aelphy) [inline test/correctness/simd_op_check_xtensa.cpp:147]: Why should it have failed?

File: test/correctness/simd_op_check.h [hunk 2]
```diff
@@ -291,12 +291,16 @@ class SimdOpCheckTest {
         return {op, error_msg.str()};
     }
 
+    void sanitize(std::string &s) {
```
[the remark this answers is not in the document]
Comment (vksnk) [reply test/correctness/simd_op_check.h:294]: I wouldn't modify it in place, I think making copy and modifying it instead and then returning from the function is preferable.

File: test/correctness/simd_op_check_xtensa.cpp [hunk 3]
```diff
@@ -79,34 +82,34 @@ class SimdOpCheckXtensa : public SimdOpCheckTest {
...
 
-        // check("halide_xtensa_widen_add_i48", vector_width / 2, i32(i16_1) + i32(i16_2));
-        // check("halide_xtensa_widen_add_u48", vector_width / 2, u32(u16_1) + u32(u16_2));
+        check("halide_xtensa_widen_add_i48", vector_width / 2, i32(i16_1) + i32(i16_2));
+        check("halide_xtensa_widen_add_u48", vector_width / 2, u32(u16_1) + u32(u16_2));
 
         // Multiplications.
         check("IVP_MULNX16PACKL", vector_width / 2, i16_1 * i16_2);
-        check("IVP_PACKLN_2X64W", vector_width / 4, i32_1 * i32_2);
+        check("IVP_PACKLN_2X64W", vector_width / 2, i32_1 * i32_2);
```
Comment (vksnk) [inline test/correctness/simd_op_check_xtensa.cpp:96]: I'd expect it to generate IVP_MULN_2X32 or something like that.

File: test/correctness/simd_op_check_xtensa.cpp (hunk 1 above)
Comment (vksnk) [reply test/correctness/simd_op_check_xtensa.cpp:147]: I think I was trying to test that it's actually will fail, if that string is not found and forgot to revert it back.

**Review by vksnk — APPROVED** [review]

**Pull request merged**